behaviour/graql/reasoner/resolution/value-predicate.feature

```diff
@@ -0,0 +1,843 @@
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: in a rule, `not { $x == $y; }` is the same as saying `$x !== $y`
+    Given for each session, graql define
+      """
+      define
+      tesco-sells-all-soft-drinks sub rule,
+      when {
+        $x isa soft-drink;
+      },
+      then {
+        $x has retailer 'Tesco';
+      };
+
+      if-ocado-exists-it-sells-all-soft-drinks sub rule,
+      when {
+        $x isa retailer;
+        $x == 'Ocado';
+        $y isa soft-drink;
+      },
+      then {
+        $y has retailer 'Ocado';
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+      $x isa soft-drink, has name "Fanta", has ref 0;
+      $y isa soft-drink, has name "Tango", has ref 1;
+      $r "Ocado" isa retailer;
+      """
+    When materialised keyspace is completed
+    Given for graql query
+      """
+      match
+        $x has retailer $r;
+        $r !== "Ocado";
+      get;
+      """
+    Given in reasoned keyspace, all answers are correct
+    # x     | r     |
+    # Fanta | Tesco |
+    # Tango | Tesco |
+    Given in reasoned keyspace, answer size is: 2
+    Then for graql query
+      """
+      match
+        $x has retailer $r;
+        not { $r == "Ocado"; };
+      get;
+      """
+    Then in reasoned keyspace, all answers are correct
+    Then in reasoned keyspace, answer size is: 2
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: in a rule, `not { $x !== $y; }` is the same as saying `$x == $y`
+    Given for each session, graql define
+      """
+      define
+      tesco-sells-all-soft-drinks sub rule,
+      when {
+        $x isa soft-drink;
+      },
+      then {
+        $x has retailer 'Tesco';
+      };
+
+      if-ocado-exists-it-sells-all-soft-drinks sub rule,
+      when {
+        $x isa retailer;
+        $x == 'Ocado';
+        $y isa soft-drink;
+      },
+      then {
+        $y has retailer 'Ocado';
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+      $x isa soft-drink, has name "Fanta", has ref 0;
+      $y isa soft-drink, has name "Tango", has ref 1;
+      $r "Ocado" isa retailer;
+      """
+    When materialised keyspace is completed
+    Given for graql query
+      """
+      match
+        $x has retailer $r;
+        $r == "Ocado";
+      get;
+      """
+    Given in reasoned keyspace, all answers are correct
+    # x     | r     |
+    # Fanta | Ocado |
+    # Tango | Ocado |
+    Given in reasoned keyspace, answer size is: 2
+    Then for graql query
+      """
+      match
+        $x has retailer $r;
+        not { $r !== "Ocado"; };
+      get;
+      """
+    Then in reasoned keyspace, all answers are correct
+    Then in reasoned keyspace, answer size is: 2
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: a negation can filter out variables by equality to another variable with a specified value
```

Comment:
Negation tests here feel more like testing negation that predicates (more advanced feature) -- lets put them in a negation feature file

Reply:
Ok I'll leave a TODO for now.